apps/bookmark/js/bookmark_editor.js

```diff
@@ -120,9 +120,37 @@ var BookmarkEditor = {
     var title = this.bookmarkTitle.value.trim();
     this.saveButton.disabled = title === '';
   },
-  
+
   _installApp: function bookmarkEditor_installApp() {
-    window.navigator.mozApps.install(this.manifestURL);
+    var request = window.navigator.mozApps.install(this.manifestURL);
+
+    request.onerror = (evt) => {
+      // Display the error information from the DOMError object
+      var reason = evt.target.error.name;
+      switch(reason) {
+        case 'INVALID_MANIFEST':
+        case 'MANIFEST_URL_ERROR':
+        case 'MANIFEST_PARSE_ERROR':
+        case 'MULTIPLE_APPS_PER_ORIGIN_FORBIDDEN':
```

Comment:
Might be useful to tell the developer specifically what went wrong in console.error?